### PR TITLE
Onboarding: Add create homepage logic to "Customize Appearance" step

### LINKED ISSUE #2481: Onboarding - Task List: Homepage Template
During the Customize Appearance section of the task list, there's an option to customize the homepage which uses a default block template.

Here's a rough template for use ( https://cloudup.com/cg0s44X1Cua ).  Note that the images should be replaced with free stock images, but we need to verify any licensing issues.  p1561034262028900-slack-wc-onboarding

This task will involve importing that block template to a new page and setting that page as the homepage if this step is completed.

Screen Shot 2019-06-21 at 4 33 06 PM

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #2481

Adds the "Create homepage" logic and template.

As discussed on Slack, the return to task list link should be displayed in a notice, however, no scripts from wc-admin are currently run on the edit post page.

### Screenshots

Screen Shot 2019-09-05 at 9 12 51 PM

### Detailed test instructions:

1. Go to the "Customize Appearance" step in the task list.
2. Click "Create homepage."
3. Publish the created homepage.
4. Navigate to any other page and note that you're redirected to the task list. (In a follow-up this will instead be a link in a notice to return to the task list instead of a redirect).
5. Return to the task list and make sure the homepage step is no longer showing until you delete the onboarding created homepage.

### CODE REVIEW

**Comment by justinshreve** [comment]
@joshuatf I almost forgot to run the unit tests here, but it looks like `test_create_homepage` is failing both locally and on travis.

File: src/API/OnboardingTasks.php [hunk 1]
```diff
@@ -134,4 +161,31 @@ public static function get_header_mappings( $file ) {
...
+			)
+		);
+
+		if ( ! is_wp_error( $post_id ) ) {
+			update_option( 'woocommerce_onboarding_homepage_post_id', $post_id );
+
+			return array(
+				'status'         => 'success',
+				'message'        => __( 'Homepage created successfully.' ),
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply src/API/OnboardingTasks.php]: Added- thanks!

File: src/API/OnboardingTasks.php [hunk 2]
```diff
@@ -65,6 +78,20 @@ public function import_products_permission_check( $request ) {
...
 
+	/**
+	 * Check if a given request has access to create a product.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function create_homepage_permission_check( $request ) {
+		if ( ! wc_rest_check_post_permissions( 'page', 'create' ) || ! current_user_can( 'manage_options' ) ) {
+			return new \WP_Error( 'woocommerce_rest_cannot_create', __( 'Sorry, you are not allowed to create a new homepage.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply src/API/OnboardingTasks.php]: Added!

File: client/dashboard/task-list/tasks/appearance.js [hunk 3]
```diff
@@ -119,12 +121,32 @@ class Appearance extends Component {
 			} );
 	}
 
+	createHomepage() {
+		const { createNotice } = this.props;
+		this.setState( { isPending: true } );
+
+		apiFetch( { path: '/wc-admin/v1/onboarding/tasks/create_homepage', method: 'POST' } )
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/task-list/tasks/appearance.js]: Replaced.

File: src/API/OnboardingTasks.php [hunk 4]
```diff
@@ -134,4 +161,31 @@ public static function get_header_mappings( $file ) {
...
...\"align\":\"\",\"backgroundColor\":\"light-gray\",\"mediaPosition\":\"right\",\"mediaId\":1257,\"mediaType\":\"image\"} -->\n<div class=\"wp-block-media-text has-media-on-the-right has-light-gray-background-color\"><figure class=\"wp-block-media-text__media\"><img src=\"https://local.wordpress.test/wp-content/uploads/2017/05/brady-bellini-191086-1024x616.jpg\" alt=\"\" class=\"wp-image-1257\"/></figure><div class=\"wp-block-media-text__content\"><!-- wp:paragraph {\"align\":\"center\",\"placeholder\":\"Content…\",\"fontSize\":\"large\"} -->\n<p style=\"text-align:center\" class=\"has-large-font-size\">A final business goal</p>\n<!-- /wp:paragraph -->\n\n<!-- wp:paragraph {\"align\":\"center\"} -->\n<p style=\"text-align:center\">Describe your business aspiration here.</p>\n<!-- /wp:paragraph --></div></div>\n<!-- /wp:media-text -->\n\n<!-- wp:woocommerce/featured-product {\"editMode\":false,\"productId\":2567} -->\n<!-- wp:button {\"align\":\"center\"} -->\n<div class=\"wp-block-button aligncenter\"><a class=\"wp-block-button__link\" href=\"https://local.wordpress.test/shop/decor/wordpress-pennant\">Shop now</a></div>\n<!-- /wp:button -->\n<!-- /wp:woocommerce/featured-product -->"
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply src/API/OnboardingTasks.php]: I think they're local images from @jameskoster's WP install.  We had some discussion before about replacing these with an online image service, but I think we were waiting on approval from legal.  @jameskoster is probably more in the know on this.

Added an issue for this [here](https://github.com/woocommerce/woocommerce-admin/issues/2902) so we can merge this PR.

[the remark this answers is not in the document]
Comment (joshuatf) [reply src/API/OnboardingTasks.php]: Also added a `@todo` comment next to the homepage template `post_content`.

**Pull request merged**